src/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -204,6 +171,42 @@ void activity_classifier::init_options(
   add_or_update_state(flexmap_to_varmap(options.current_option_values()));
 }
 
+// Randomly split an SFrame into two SFrames based on the `session_id` such
```

Comment:
Nit: In this case, I think we may as well have the static function come first in this file, just after the anonymous namespace. This way, it doesn't break up the sequence of instance methods.